siriuspy/siriuspy/devices/ids.py

```diff
@@ -1278,3 +1278,30 @@ def __init__(self, devname=None, props2init='all', auto_monitor_mon=True):
         # call base class constructor
         super().__init__(
             devname, props2init=props2init, auto_monitor_mon=auto_monitor_mon)
+
+
+class IDFactory():
```

Comment:
We don't use this strategy of creating a factory in the devices package. See the implementation for SOFB, for instance. There, the class SOFB instantiates the appropriate class, depending on the devname supplied. I think here you could have an `ID` or `InsertionDevice`, that instantiates the appropriate object, instead of an `IDFactory` class, with an `staticmethod` `create` to do so.

Reply:
who are "we" ? I just used this strategy! :smile:

I looked at SOFB now. You propose to use a general device class to instantiate specialized device class objects, instead of using a non-device class for this purpose, right ? Ok, it sounds an alternative possibility (but I see no particular advantage other than using what has been done already). I will call this class `class IDfactory(_Device)`... kidding.

Reply:
changed. take a look.

Reply:
Nice, @xresende, thanks. We are all of us, I think. Because all devices  and devices set up to now have a standardized way of instantiation and the factory strategy breaks it, making the library less intuitive.